integration_tests/src/main/python/hash_aggregate_test.py

```diff
@@ -697,10 +698,10 @@ def do_it(spark):
         df = gen_df(spark, data_gen, length=100)\
             .groupby('a')\
             .agg(f.collect_set('b').alias("collect_set"))
-        # pull out the rdd and schema and create a new dataframe to run SortArray
+        # collect the data and schema and create a new dataframe to run SortArray
         # to handle Spark 3.3.0+ optimization that moves SortArray from ProjectExec
         # to ObjectHashAggregateExec
-        return spark.createDataFrame(df.rdd, schema=df.schema)\
+        return spark.createDataFrame(df.collect(), schema=df.schema)\
```

Comment:
let us think if this is really the best we can do as it creates quite an overhead.

Reply:
I've honestly never fully liked this solution anyways (when it was recreating a dataframe and sorting it using the CPU). Thinking about it, it might just be better to do this sort in Python and add functionality to `@ignore_order` for it. Thoughts? It might even be cheaper (for the most part) than doing the whole thing is Spark for <3.3 because that still has to do the sort in CPU Spark anyways.

Reply:
Pushed a possible solution to sort the arrays in Python instead of sending it back through Spark. Let me know if this makes more sense.